test/tests.cpp

```diff
@@ -371,23 +371,40 @@ TEST_CASE_METHOD(QInterfaceTestFixture, "test_swap")
 {
     qftReg->SetPermutation(0xb2000);
     qftReg->Swap(12, 16, 4);
-    REQUIRE_THAT(qftReg, HasProbability(0, 8, 0x2b000));
```

Comment:
I want to see a bunch of tests here that exercise the new code - please validate with code coverage numbers that the coverage is comprehensive.

Reply:
You can see I added unit tests for the new public API methods. The phase gate optimizations in QUnit are entirely internal and covered by the original controlled phase gate unit tests.